spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/config/ErrorChannelTests.java

```diff
@@ -0,0 +1,140 @@
+import org.springframework.integration.core.MessageSource;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.MessagingException;
+import org.springframework.messaging.SubscribableChannel;
+import org.springframework.messaging.support.ErrorMessage;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.util.Assert;
+
+/**
+ * @author Ilayaperumal Gopinathan
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration({ErrorChannelTests.TestSource.class, ErrorChannelTests.TestErrorSink1.class,
+		ErrorChannelTests.TestErrorSink2.class})
+public class ErrorChannelTests {
+
+	@Rule
+	public RedisTestSupport redisTestSupport = new RedisTestSupport();
+
+	@Autowired
+	@Bindings(TestErrorSink1.class)
+	private ErrorSink1 testErrorSink1;
+
+	@Autowired
+	@Bindings(TestErrorSink2.class)
+	private ErrorSink2 testErrorSink2;
+
+	@Test
+	public void testErrorChannelBinding() throws Exception {
+		final CountDownLatch latch = new CountDownLatch(1);
+		MessageHandler errorMessageHandler = new MessageHandler() {
+			@Override
+			public void handleMessage(Message<?> message) throws MessagingException {
+				Assert.isTrue(message instanceof ErrorMessage, "Message should be an instance of ErrorMessage");
+				Assert.isTrue(message.getPayload() instanceof MessagingException, "Message payload should be an instance" +
+						"of MessagingException");
+				Assert.isTrue(message.getPayload().toString()
+						.equals("org.springframework.messaging.MessagingException: test"));
+				latch.countDown();
+			}
+		};
+		testErrorSink1.errorInput().subscribe(errorMessageHandler);
+		testErrorSink2.errorInput().subscribe(errorMessageHandler);
+		assertTrue(latch.await(10, TimeUnit.SECONDS));
+	}
+
+	@EnableBinding(Source.class)
+	@EnableAutoConfiguration
+	@Import(RedisMessageChannelBinderConfiguration.class)
+	@PropertySource("classpath:/org/springframework/cloud/stream/config/errorchannel/source-channel.properties")
+	public static class TestSource {
+
+		@Bean
+		@InboundChannelAdapter(value = Source.OUTPUT, poller = @Poller(fixedDelay = "5000", maxMessagesPerPoll = "1"))
+		public MessageSource<String> timerMessageSource() {
+			return new MessageSource<String>() {
+				@Override
+				public Message<String> receive() {
+					throw new MessagingException("test");
+				}
+			};
+		}
+	}
+
+	@EnableBinding(ErrorSink1.class)
+	@EnableAutoConfiguration
+	@Import(RedisMessageChannelBinderConfiguration.class)
+	@PropertySource("classpath:/org/springframework/cloud/stream/config/errorchannel/errorsink1-channel.properties")
+	public static class TestErrorSink1 {
+
+	}
+
+	public interface ErrorSink1 {
+
+		String INPUT = "errorInput1";
+
+		@Input(ErrorSink1.INPUT)
+		SubscribableChannel errorInput();
+
+	}
+
+	@EnableBinding(ErrorSink2.class)
+	@EnableAutoConfiguration
+	@Import(RedisMessageChannelBinderConfiguration.class)
```

Comment:
I agree with @mbogoevici. They are "integration" tests as long as any binder is involved. The `TestBinder` *must* be sufficient for such tests, so that any binder would be guaranteed to work - as long as each binder implementation satisfies the `AbstractBinderTest`-based "compatibility test" suite (which should probably become more formalized with time).

Furthermore, even if do continue to host binder implementations, such as Kafka and Rabbit, within this repository, we know that other binder implementations will exist outside of the scope of this repository. In theory, (this will take some time to achieve but should a goal IMO), we should be able to move all but the `TestBinder` out of this repository and not have any dependencies on other binder(s).